src/fmeval/eval_algorithms/summarization_accuracy.py

```diff
@@ -150,72 +171,45 @@ def evaluate(
+            "for the evaluate method to run successfully.",
+        )
         dataset_configs = get_dataset_configs(dataset_config, self.eval_name)
         eval_outputs = []
         for dataset_config in dataset_configs:
-            dataset = get_dataset(dataset_config, num_records)
-            validate_dataset(dataset, [DatasetColumns.TARGET_OUTPUT.value.name, DatasetColumns.MODEL_INPUT.value.name])
-            dataset_prompt_template = None
-            pipeline = self.pipeline
-
-            if DatasetColumns.MODEL_OUTPUT.value.name not in dataset.columns():
-                require(model, "No ModelRunner provided. ModelRunner is required for inference on model_inputs")
-                dataset_prompt_template = (
-                    get_default_prompt_template(dataset_config.dataset_name) if not prompt_template else prompt_template
-                )
-                gen_prompt = GeneratePrompt(
-                    input_keys=[DatasetColumns.MODEL_INPUT.value.name],
-                    output_keys=[DatasetColumns.PROMPT.value.name],
-                    prompt_template=dataset_prompt_template,
-                )
-                get_model_outputs = GetModelOutputs(
-                    input_to_output_keys={DatasetColumns.PROMPT.value.name: [DatasetColumns.MODEL_OUTPUT.value.name]},
-                    model_runner=model,
-                )
-                pipeline = TransformPipeline([gen_prompt, get_model_outputs, pipeline])
-
-            with timed_block(f"Computing score and aggregation on dataset {dataset_config.dataset_name}", logger):
-                dataset = pipeline.execute(dataset)
-                dataset_scores, category_scores = aggregate_evaluation_scores(dataset, METRIC_NAMES, agg_method=MEAN)
-                eval_outputs.append(
-                    EvalOutput(
-                        eval_name=self.eval_name,
-                        dataset_name=dataset_config.dataset_name,
-                        prompt_template=dataset_prompt_template,
-                        dataset_scores=dataset_scores,
-                        category_scores=category_scores,
-                        output_path=generate_output_dataset_path(
-                            path_to_parent_dir=self._eval_results_path,
-                            eval_name=self.eval_name,
-                            dataset_name=dataset_config.dataset_name,
-                        ),
-                    )
-                )
-            if save:
-                save_dataset(
-                    dataset=dataset,
-                    score_names=METRIC_NAMES,
-                    path=generate_output_dataset_path(
-                        path_to_parent_dir=self._eval_results_path,
-                        eval_name=self.eval_name,
-                        dataset_name=dataset_config.dataset_name,
-                    ),
-                )
-
+            eval_output = util.evaluate_dataset(
```

Comment:
I find it confusing that the function that brings together everything (i.e., running a pipeline on a dataset) is called `evaluate_dataset`, could it be called `run_pipeline` or similar instead? Or alternatively, could this be split into two helpers: `prepare_dataset` and `compute_and_aggregate_metrics` perhaps?

Reply:
I actually originally had two function (the second one was literally named `compute_and_aggregate_metrics`, just as you suggested), but after various refactoring, `evaluate_dataset` was basically just calling `compute_and_aggregate_metrics`, so I subsumed it.